server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/PdbDataManager.java

```diff
@@ -46,82 +56,149 @@
-            }
+                    return parseAlignment(pdbBlocks);
+                }).collect(toList());
 
-            return fetchedDasalignmentData;
-        } catch (JAXBException e) {
-            throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
-                    .ERROR_UNEXPECTED_FORMAT, location), e);
+        return parseDasalignment(alignments);
+    }
+
+    private Dataset parseToDataset(DatasetDTO datasetDTO, String structureId) {
+        Entry entry = datasetDTO.getData().getEntry();
+        String title = entry.getStruct().getTitle();
+        String classification = entry.getKeywords().getValue();
+
+        List<Record> records = entry.getEntities().stream().flatMap(entity -> {
+            String compound = entity.getRcsbPoly().getDescription();
+            String uniName = validateUniName(entity);
+
+            return entity.getIdentifiers().getIds().stream()
+                    .map(chainId -> parseRecord(title, classification, compound, uniName, structureId, chainId))
+                    .collect(toList())
+                    .stream();
+
+        }).collect(toList());
+
+        return parseDataset(records);
+    }
+
+    private String validateUniName(PolymerEntity entity) {
+        List<PolymerEntity.Uniprot> uniprots = entity.getUniprots();
+        return (uniprots == null || uniprots.isEmpty()) ? "" : uniprots.get(0).getNames().get(0);
+    }
+
+    private Record parseRecord(String title, String classification, String compound,
+                               String uniprotName, String structureId, String chainId) {
+        Record record = new Record();
+        record.setStructureTitle(title);
+        record.setClassification(classification);
+        record.setCompound(compound);
+        record.setUniprotRecommendedName(uniprotName);
+        record.setStructureId(structureId);
+        record.setChainId(chainId);
+        return record;
+    }
+
+    private Dataset parseDataset(List<Record> records) {
+        Dataset dataset = new Dataset();
+        dataset.setRecord(records);
+        return dataset;
+    }
+
+    private Segment parseSegment(Integer start, Integer end, String id) {
+        //start and end coordinates are the same, if no end.
+        if (end == null) {
+            end = start;
         }
+        Segment segment = new Segment();
+        segment.setStart(start);
+        segment.setEnd(end);
+        segment.setIntObjectId(id);
+        return segment;
+    }
+
+    private PdbBlock parsePdbBlock(List<Segment> segments) {
+        PdbBlock block = new PdbBlock();
+        block.setSegment(segments);
+        return block;
+    }
+
+    private Alignment parseAlignment(List<PdbBlock> blocks) {
+        Alignment alignment = new Alignment();
+        alignment.setBlock(blocks);
+        return alignment;
+    }
+
+    private Dasalignment parseDasalignment(List<Alignment> alignments) {
```

Comment:
final. Please check all the code for such things

Reply:
fixed!